packages/lib-ts/src/types/EvmEncodeParam.ts

```diff
@@ -1,27 +1,22 @@
-import { join, Serializable, serialize, serializeArray, Stringable } from '../helpers'
+import { serialize, Stringable } from '../helpers'
+
+@json
+export abstract class EvmEncodeParamBase {
+  constructor(public readonly abiType: string) {}
+}
 
 /**
  * Represents a parameter for EVM encoding, used to prepare data for smart contract calls.
  * This class encapsulates both primitive values and complex structures (arrays, tuples)
  * in a format suitable for ABI encoding.
  */
-export class EvmEncodeParam implements Serializable {
-  private static readonly SERIALIZED_PREFIX: string = 'EvmEncodeParam'
-
-  private _type: string
-  private _value: string
-  private _values: EvmEncodeParam[]
-
-  /**
-   * Creates a new EvmEncodeParam instance.
-   * @param type - The ABI type signature (e.g., "uint256", "address", "string", "()")
-   * @param value - The serialized value for primitive types
-   * @param values - Array of nested EvmEncodeParam instances for complex types
-   */
-  constructor(type: string, value: string, values: EvmEncodeParam[]) {
-    this._type = type
-    this._value = value
-    this._values = values
+@json
+export class EvmEncodeParam extends EvmEncodeParamBase {
```

Comment:
The ultimate goal is to send the relayer a JSON equivalent to this:

```typescript
@json
export class EvmEncodeParam {
  constructor(
    public readonly abiType: string,
    public readonly value: string | EvmEncodeParam[]
  )
}
```

Due to the restriction of AssemblyScript not implementing union types, the way to get around this is to use this inheritance pattern, which is a bit tricky in code but maintains strong typing.